src/test/java/org/kohsuke/github/GHRepositoryForkBuilderTest.java

```diff
@@ -0,0 +1,235 @@
+package org.kohsuke.github;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InterruptedIOException;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.*;
+import static org.mockito.Mockito.*;
+
+// TODO: Auto-generated Javadoc
+
+/**
+ * The Class GHRepositoryForkBuilderTest.
+ */
+public class GHRepositoryForkBuilderTest extends AbstractGitHubWireMockTest {
+    private GHRepository repo;
+    private static final String TARGET_ORG = "nts-api-test-org";
+
+    /**
+     * Instantiates a new Gh repository fork builder test.
+     */
+    public GHRepositoryForkBuilderTest() {
+    }
+
+    /**
+     * Sets up.
+     *
+     * @throws Exception
+     *             the exception
+     */
+    @Before
+    public void setUp() throws Exception {
+        repo = getTempRepository();
+
+        String defaultBranch = repo.getDefaultBranch();
+        GHRef mainRef = repo.getRef("heads/" + defaultBranch);
+        String mainSha = mainRef.getObject().getSha();
+
+        String[] branchNames = { "test-branch1", "test-branch2", "test-branch3" };
+        for (String branchName : branchNames) {
+            repo.createRef("refs/heads/" + branchName, mainSha);
+        }
+    }
+
+    private void verifyBasicForkProperties(GHRepository original, GHRepository forked, String expectedName)
+            throws IOException, InterruptedException {
+        GHRepository updatedFork = forked;
+        for (int i = 0; i < 10; i++) {
+            Thread.sleep(3000);
+            updatedFork = gitHub.getRepository(forked.getFullName());
+            if (updatedFork.isFork()) {
+                break;
+            }
+        }
```

Comment:
Use `awaitility` instead of manually sleep calls.  When running in non-snapshot recording mode, the sleeps can be skipped.